guides/plugins/plugins/framework/data-handling/field-inheritance.md

```diff
@@ -0,0 +1,227 @@
+     */
+    protected $parentId;
+
+    /**
+     * @var ExampleCollection|null
+     */
+    protected $children;
+
+    ...
+
+    public function getParent(): ?ExampleEntity
+    {
+        return $this->parent;
+    }
+
+    public function setParent(ExampleEntity $parent): void
+    {
+        $this->parent = $parent;
+    }
+
+    public function getParentId(): ?string
+    {
+        return $this->parentId;
+    }
+
+    public function setParentId(?string $parentId): void
+    {
+        $this->parentId = $parentId;
+    }
+
+    public function getChildren(): ?ExampleCollection
+    {
+        return $this->children;
+    }
+
+    public function setChildren(ExampleCollection $children): void
+    {
+        $this->children = $children;
+    }
+}
+```
+{% endcode %}
+
+## Translations
+
+This concept also supports translations. Given a parent/child entity with an
+inherited language (de-CH *inherits from* de-DE), the resolution of the
+values will be:
+
+1. Child (de-CH)
+2. Child (de-DE)
+3. Parent (de-CH)
+4. Parent (de-DE)
+
+If an inheritance is not found, the next translation in the chain above will be used.
+
+### Enable translation inheritance
+
+Assuming our definition is already aware of inheritance, we have to update 
+our definition and add the `Inherited` flag to our translated fields and
+the translation association.
+
+```php
+(new TranslatedField('name'))->addFlags(new Inherited()),
+(new TranslationsAssociationField(ExampleTranslationDefinition::class))->addFlags(new Inherited()),
+```
+
+## Add inheritance via plugin
+
+If you want to add an inherited association to an entity, the following must be considered:
+
+1. the entity to be extended must already fulfill the above mentioned conditions
+2. you have to add the appropriate association via your EntityExtension class.
+3. the new association requires the `Inherited` flag
+4. you need to add a column in the MySQL table of the entity that has the following properties:
+    * binary(16)
+    * nullable
+    * default null
+
+To avoid creating the column incorrectly, you can simply use the `Shopware\Core\Framework\Migration\InheritanceUpdaterTrait` in your migrations:
+
+{% code title="<plugin root>/src/Migration/Migration1615363012AddInheritanceColumnToExample.php" %}
+```php
+<?php declare(strict_types=1);
+
+namespace Swag\BasicExample\Migration;
+
+use Doctrine\DBAL\Connection;
+use Shopware\Core\Framework\Migration\InheritanceUpdaterTrait;
+use Shopware\Core\Framework\Migration\MigrationStep;
+
+class Migration1615363012AddInheritanceColumnToExample extends MigrationStep
+{
+    use InheritanceUpdaterTrait;
+
+    public function getCreationTimestamp(): int
+    {
+        return 1615363012;
+    }
+
+    public function update(Connection $connection): void
+    {
+        $this->updateInheritance($connection, 'swag_example', 'example_field');
+    }
+
+    public function updateDestructive(Connection $connection): void
+    {
+    }
+}
+```
+{% endcode %}
```

Comment:
Why are migrations at the end of the document ? 
Why is not in the bullet list on the top of the file ?